conda_forge_tick/git_utils.py

```diff
@@ -225,63 +254,114 @@ def _run_git_command(
 
     @lock_git_operation()
     def add(self, git_dir: Path, *pathspec: Path, all_: bool = False):
-        """
-        Add files to the git index with `git add`.
-        :param git_dir: The directory of the git repository.
-        :param pathspec: The files to add.
-        :param all_: If True, not only add the files in pathspec, but also where the index already has an entry.
-        If all_ is set with empty pathspec, all files in the entire working tree are updated.
-        :raises ValueError: If pathspec is empty and all_ is False.
-        :raises GitCliError: If the git command fails.
+        """Add files to the git index with `git add`.
+
+        Parameters
+        ----------
+        git_dir : str
+            The directory of the git repository.
+        pathspec : str
+            The files to add.
+        all_ : bool, optional
+            If True, not only add the files in pathspec, but also where the index
+            already has an entry. If all_ is set with empty pathspec, all files
+            in the entire working tree are updated.
+
+        Raises
+        ------
+        ValueError
+            If pathspec is empty and all_ is False.
+        GitCliError
+            If the git command fails.
         """
         if not pathspec and not all_:
             raise ValueError("Either pathspec or all_ must be set.")
 
         all_arg = ["--all"] if all_ else []
 
-        self._run_git_command(["add", *all_arg, *pathspec], git_dir)
+        try:
+            self._run_git_command(["add", *all_arg, *pathspec], git_dir)
+        except GitCliError as e:
+            raise GitCliError("Adding files to git failed.") from e
```

Comment:
Same here.

Reply:
This change is required because of the new `DOC501` and `DOC502` ruff rules. It doesn't change any behaviour, just makes existing errors more explicit.